### PR TITLE
feat:[SRM-15506]:Implemented environment based and composite SLO base…

### PR DESCRIPTION
**Pull request by shash-harness**
…d filter for the SLO listing

## Describe your changes
Added the code for filtering the SLO based on list of environment Identifiers and also on basis of whether we want compositeSLO only or not. Also added an api to get the dropdown for environments.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52879)

### CODE REVIEW

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Review by deepak-harness — CHANGES_REQUESTED** [review]
add testcases

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52879#-:-NeGv3lt7J5M-XMfYpKX:bsxjolp)* status: 6 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/beans/EnvIdentifiersDropdownResponse.java` line 16 at r2](https://reviewable.io/reviews/harness/harness-core/52879#-NeGt_GGCRy7rpjCQVF7:-NeGt_GGCRy7rpjCQVF8:b-gssdjc) ([raw file](https://github.com/harness/harness-core/blob/ad4a8f265ef515057f7a632610f6c895299ba2a6/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/beans/EnvIdentifiersDropdownResponse.java#L16)):*
> ```Java
> @Value
> @Builder
> public class EnvIdentifiersDropdownResponse {
> ```

EnvironmentIdentifierResponse
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImpl.java` line 754 at r2](https://reviewable.io/reviews/harness/harness-core/52879#-NeGuSAX5xi8bK4pIbtf:-NeGuSAX5xi8bK4pIbtg:bndip4t) ([raw file](https://github.com/harness/harness-core/blob/ad4a8f265ef515057f7a632610f6c895299ba2a6/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImpl.java#L754)):*
> ```Java
>                     .getName();
>     } catch (Exception e) {
>       envName = "DELETED";
> ```

log. warning("envirionment with <identifier> not present")

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/beans/EnvIdentifiersDropdownResponse.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+package io.harness.cvng.servicelevelobjective.beans;
+
+import javax.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class EnvIdentifiersDropdownResponse {
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/beans/EnvIdentifiersDropdownResponse.java:16]: EnvIdentifier* (should be singular) ?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImpl.java [hunk 2]
```diff
@@ -717,6 +741,21 @@ private MSDropdownResponse getMSDropdownResponse(MonitoredServiceDTO monitoredSe
...
+      ProjectParams projectParams, MonitoredServiceDTO monitoredServiceDTO) {
+    String envIdentifier = monitoredServiceDTO.getEnvironmentRef();
+    String envName;
+    try {
+      envName = nextGenService
+                    .getEnvironment(projectParams.getAccountIdentifier(), projectParams.getOrgIdentifier(),
+                        projectParams.getProjectIdentifier(), envIdentifier)
+                    .getName();
+    } catch (Exception e) {
+      envName = "DELETED";
```
Comment (naveen-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/SLODashboardServiceImpl.java:754]: please add debug or warning log as well here.

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger ut7

**Comment by shash-harness** [comment]
trigger uts

**Review by deepak-harness — APPROVED** [review]
> Reviewed 2 of 6 files at r1, 4 of 4 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52879)* status: :shipit: complete! ✓

**Pull request merged**